src/pd/async/util.rs

```diff
@@ -171,3 +185,153 @@ impl<Req, Resp, F> Request<Req, Resp, F>
             .boxed()
     }
 }
+
+/// Do a request in synchronized fashion.
+pub fn sync_request<F, R>(client: &LeaderClient, f: F, retry: usize) -> Result<R>
```

Comment:
This function waits request futures, in a caller's perspective, it does a synchronous request, so I add a prefix `sync`.